File: Languages/C#/Day025.cs
// Languages - C# - Day 25
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that converts a Decimal Number to Binary without using built-in conversion functions.


using System;
using System.Text;

class Program
{
    static string DecimalToBinary(int n)
    {
        if (n == 0) return "0";

        bool isNegative = n < 0;
        int num = Math.Abs(n);
        StringBuilder binary = new StringBuilder();

        while (num > 0)
        {
            binary.Insert(0, num % 2);
            num /= 2;
        }

        if (isNegative)
        {
            binary.Insert(0, "-");
        }

        return binary.ToString();
    }

    static void Main()
    {
        Console.Write("Enter a decimal number: ");
        string input = Console.ReadLine()?.Trim();

        if (!int.TryParse(input, out int number))
        {
            Console.WriteLine("Invalid input. Please enter an integer.");
            return;
        }

        Console.WriteLine("Binary representation: " + DecimalToBinary(number));
    }
}

File: Languages/Java/Day025.java
// Languages - Java - Day 25
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that converts a Decimal Number to Binary without using built-in conversion functions.


import java.util.Scanner;

public class Day025 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a decimal number: ");
        String input = scanner.nextLine().trim();

        try {
            int number = Integer.parseInt(input);
            String binary = decimalToBinary(number);
            System.out.println("Binary representation: " + binary);
        } catch (NumberFormatException e) {
            System.out.println("Invalid input. Please enter a valid integer.");
        }

        scanner.close();
    }

    public static String decimalToBinary(int number) {
        if (number == 0) {
            return "0";
        }

        int num = Math.abs(number);
        StringBuilder binary = new StringBuilder();
        
        while (num > 0) {
            binary.insert(0, num % 2);
            num /= 2;
        }

        return number < 0 ? "-" + binary.toString() : binary.toString();
    }
}
